impl/events.go

```diff
@@ -32,10 +32,14 @@ func (m *manager) OnDataReceived(chid datatransfer.ChannelID, link ipld.Link, si
 	if chid.Initiator != m.peerID {
 		var result datatransfer.VoucherResult
 		var err error
+		var handled bool
 		_ = m.revalidators.Each(func(_ datatransfer.TypeIdentifier, _ encoding.Decoder, processor registry.Processor) error {
 			revalidator := processor.(datatransfer.Revalidator)
-			result, err = revalidator.OnPushDataReceived(chid, size)
-			return err
+			handled, result, err = revalidator.OnPushDataReceived(chid, size)
+			if handled {
+				return errors.New("stop processing")
```

Comment:
Why do we need to return this error ? So that we break the `m.revalidators.Each` loop ?

Is there any other significance of this error ?

Reply:
yea that's what it does -- it stops the loop -- note that it's ignored in the return